src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,309 @@
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        greeting();
+        loadTasks();
+        getCommand();
+    }
+
+    private static void getCommand() {
+        String input;
+        Scanner in = new Scanner(System.in);
+        // get user input
+        input = in.nextLine();
+        while (!input.equals("bye")) {
+            if (input.equals("list")) {
+                showList();
+                input = in.nextLine();
+                continue;
+            }
+            if (input.startsWith("done")) {
+                int number = Integer.parseInt(input.substring(input.length() - 1));
+                markAsDone(number);
+                saveTasks();
+                input = in.nextLine();
+                continue;
+            }
+            if (input.startsWith("delete")) {
+                int number = Integer.parseInt(input.substring(input.length() - 1));
+                deleteTask(number);
+                saveTasks();
+                input = in.nextLine();
+                continue;
+            }
+            if (input.startsWith("delete")) {
+                int number = Integer.parseInt(input.substring(input.length() - 1));
+                deleteTask(number);
+                input = in.nextLine();
+                continue;
+            }
+            try {
+                checkCommand(input);
+            } catch (InvalidCommandException | EmptyDescriptionException e) {
+                input = in.nextLine();
+                continue;
+            }
+            // Command is valid, handle the command
+            System.out.println("    ____________________________________________________________");
+            System.out.println("     Got it. I've added this task: ");
+            handleCommand(input);
+            saveTasks();
+            System.out.println("     Now you have " + tasks.size() + " tasks in the list.");
+            System.out.println("    ____________________________________________________________");
+            input = in.nextLine();
+        }
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Bye. Hope to see you again soon!");
+        System.out.println("    ____________________________________________________________");
+    }
+
+    private static void checkCommand(String line) throws EmptyDescriptionException, InvalidCommandException {
+        Set<String> validCommands = Set.of("todo", "deadline", "event");
+        String[] splitLine = line.split(" ");
+        String type = splitLine[0];
+        if (!validCommands.contains(type)) {
+            throw new InvalidCommandException();
+        }
+        if (splitLine.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+    }
+
+    private static void loadTasks() {
```

Comment:
You can create a separate class for things that have to do with the storing of information into a text file, instead of including it all in Duke